red_october/models/red_october_user.py

```diff
@@ -0,0 +1,206 @@
+    name = fields.Char(
+        related='user_id.name',
+    )
+    vault_ids = fields.Many2many(
+        string='Vaults',
+        comodel_name='red.october.vault',
+        required=True,
+        default=lambda s: s._default_vault_ids(),
+        domain="[('user_id.company_id', 'in', company_ids)]",
+    )
+    delegation_ids = fields.One2many(
+        string='Delegations',
+        comodel_name='red.october.delegation',
+        inverse_name='user_id',
+    )
+    is_admin = fields.Boolean(
+        string='Admin',
+    )
+    is_active = fields.Boolean()
+    encrypt_type = fields.Selection([
+        ('RSA', 'RSA'),
+        ('ECC', 'ECC'),
+    ],
+        default='RSA',
+        required=True,
+    )
+
+    @api.model
+    def _default_vault_ids(self):
+        return [
+            (6, 0, self.env['red.october.vault'].get_current_vault().ids),
+        ]
+
+    @api.multi
+    @api.constrains('user_id', 'vault_ids')
+    def _check_user_id_vault_ids(self):
+        for record in self:
+            results = self.search([
+                ('user_id', '=', record.user_id.id),
+                ('vault_ids', 'in', record.vault_ids.ids),
+            ])
+            if len(results) > 1:
+                raise ValidationError(_(
+                    'This user is already registered in this vault.',
+                ))
+
+    @api.model
+    def create(self, vals):
+        """ It creates the user on the remote vaults. """
+        already_active = vals.get('is_active')
+        vals['is_active'] = True
+        res = super(RedOctoberUser, self).create(vals)
+        if already_active:
+            return res
+        for vault in res.vault_ids:
+            with vault.get_api() as api:
+                api.create_user()
+                if res.is_admin:
+                    record._update_role('admin')
+        res.active = True
+        return res
+
+    @api.multi
+    def update(self, vals):
+        """ It updates the user on the remote vaults. """
+        for record in self:
+            role = None
+            active = vals.get('is_active')
+            is_admin = vals.get('is_admin')
+            if is_admin is not None and is_admin != record.is_admin:
+                role = 'admin'
+            elif active is not None and active != record.active:
+                role = 'revoke'
+            if role:
+                record._update_role(role)
+        return super(RedOctoberUser, self).update(vals)
+
+    @api.multi
+    def unlink(self):
+        """ It removes the user from the remote vaults. """
+        self._update_role('revoke')
+        return super(RedOctoberUser, self).unlink()
+
+    @api.model
+    def change_current_user(self, ro_user_id):
+        """ It changes the current session user to the provided.
+
+        Args:
+            ro_user_id (int): ID of the RedOctoberUser to add as the current
+                session default.
+        """
+        setattr(request.session, self.SESSION_USER_ATTR, ro_user_id)
```

Comment:
Yes, but only in the sense that you would appear logged in as that user. You would still need to provide a decryption password into the UI - so no big deal.

The decryption password is stored in the browser memory due to the Odoo frame not actually changing as you navigate the backend. It is never stored in the session, although the user does default with the Odoo password from the session if they are not created through the Vault initialization. This weakness is then circumvented by changing the password via the Crypto Menu